tests/check/check_h26xsigned_sign.c

```diff
@@ -581,53 +581,69 @@ START_TEST(undefined_nalu_in_sequence)
 }
 END_TEST
 
-START_TEST(two_seis_in_buffer)
+/* Test description
+ * Verify that after 2 completed Sei's created ,they will be transfered in
+ * correct order
+ * The operation is as follows:
+ * 1. Setup a signed_video_t session
+ * 2. Add  2 I NALU for signing that will trigger 2 Sei Nalu
+ * 3. Get the Sei NALU's to prepend
+ * 4. Check that Sei NALU's transfered in  correct order
+ */
+START_TEST(two_completed_seis_pending)
 {
-  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
-  // |settings|; See signed_video_helpers.h.
-
   SignedVideoCodec codec = settings[_i].codec;
   SignedVideoReturnCode sv_rc;
   signed_video_nalu_to_prepend_t nalu_to_prepend = {0};
+  signed_video_nalu_to_prepend_t nalu_to_prepend_2 = {0};
+
   signed_video_t *sv = signed_video_create(codec);
+  sv->sv_test_on = true;
   ck_assert(sv);
   char *private_key = NULL;
   size_t private_key_size = 0;
   nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id('I', 0, codec);
-
+  nalu_list_item_t *i_nalu_2 = nalu_list_item_create_and_set_id('I', 0, codec);
   // Setup the key
   sv_rc =
-      signed_video_generate_private_key(settings[_i].algo, "./", &private_key, &private_key_size);
+      signed_video_generate_private_key(settings[_i].algo, NULL, &private_key, &private_key_size);
   ck_assert_int_eq(sv_rc, SV_OK);
 
   sv_rc = signed_video_set_private_key(sv, settings[_i].algo, private_key, private_key_size);
   ck_assert_int_eq(sv_rc, SV_OK);
   sv_rc = signed_video_set_authenticity_level(sv, settings[_i].auth_level);
   ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc =
+      signed_video_add_nalu_for_signing_with_timestamp(sv, i_nalu->data, i_nalu->data_size, NULL);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_add_nalu_for_signing_with_timestamp(
+      sv, i_nalu_2->data, i_nalu_2->data_size, NULL);
+  ck_assert_int_eq(sv_rc, SV_OK);
 
-  sv_rc = signed_video_add_nalu_for_signing_with_timestamp_for_test(
-      sv, i_nalu->data, i_nalu->data_size, NULL);
+  // After 2 seis are created we can copy the sei buffer
+  sei_data_t sei_buff[2];
+  memcpy(sei_buff, sv->sei_data_buffer, sizeof(sei_buff));
   ck_assert_int_eq(sv_rc, SV_OK);
   sv_rc = signed_video_get_nalu_to_prepend(sv, &nalu_to_prepend);
   ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_get_nalu_to_prepend(sv, &nalu_to_prepend_2);
+  ck_assert_int_eq(sv_rc, SV_OK);
```

Comment:
You should make a third call and verify that you get a "nothing to do" instruction
This can preferably be done after the checks below